modules/util/util.js

```diff
@@ -218,6 +218,16 @@ export function utilDisplayName(entity) {
         name = t('inspector.display_name.' + keyComponents.join('_'), tags);
     }
 
+    // if there's still no name found, try addr:housename
+    if (!name && entity.tags['addr:housename']) {
+        name = entity.tags['addr:housename'];
+    }
+
+    // as a last resort, use the street address as a name
+    if (!name && entity.tags['addr:housenumber'] && entity.tags['addr:street']) {
+        name = entity.tags['addr:housenumber'] + ' ' + entity.tags['addr:street'];
```

Comment:
This format may not be appropriate in every locale. For example, in China in Chinese, an address goes from biggest to smallest, so the street would come before the house number. In Vietnam in Vietnamese, the word “Số” would ideally precede the house number. Something like this would at least make the format localizable by language:

```suggestion
        name = t('inspector.display_name_addr', {
          housenumber: entity.tags['addr:housenumber'],
          street: entity.tags['addr:street']
        });
```

Then core.yaml would set `inspector.display_name_addr` to `{housenumber} {street}` to match the expected English format.

Unfortunately, `t()` on its own can’t resolve any country-specific differences. Perhaps this functionality could be added to address_formats.json.

Reply:
Another simpler option would be to return just `addr:housenumber`, which would be short enough to label the points on the map canvas, though it could get ambiguous in the sidebar.

Reply:
@1ec5 good point, I've made that change.

Regarding your second comment - In the future, perhaps `utilDisplayName()` could be split into `utilDisplayShortName()` and  `utilDisplayLongName()`, then the short name could be used on the map (just `addr:housenumber`) and the long name used everywhere else. But I'll leave that for a separate PR